### ISSUE TITLE
KeyVault *_as_lazy_as_lazy

### ISSUE THREAD

**Issue by jstange** [Azure/azure-sdk-for-ruby#2826]
Just started experimenting with the item-manipulating parts of the KeyVault interface (`azure_key_vault-0.18.0`), and I'm getting stuff like this:

```
NoMethodError: undefined method `get_certificates_as_lazy_as_lazy' for #<Azure::KeyVault::V7_1::KeyVaultClient:0x0000000008047da8>
Did you mean?  get_certificates_as_lazy_async
               get_certificates_as_lazy
               get_certificate_issuers_as_lazy
["/var/rubies/ruby-2.7.2/lib/ruby/gems/2.7.0/gems/azure_key_vault-0.18.0/lib/7.1/generated/azure_key_vault/key_vault_client.rb:11124:in `get_certificates_as_lazy'",
 "/var/rubies/ruby-2.7.2/lib/ruby/gems/2.7.0/gems/azure_key_vault-0.18.0/lib/7.1/generated/azure_key_vault/key_vault_client.rb:131:in `get_certificates'"
[...]               
```

The various `get_*` methods in here appear to, through some layers of indirection, call a method which doesn't exist. I see this behavior with all available client versions in here.

The culprit methods look like:

```
    def get_certificates_as_lazy(vault_base_url, maxresults:nil, include_pending:nil, custom_headers:nil)
      first_page = get_certificates_as_lazy_as_lazy(vault_base_url, maxresults:maxresults, include_pending:include_pending, custom_headers:custom_headers)
      first_page.get_all_items
    end
```

The above is identical to `get_certificates` (which calls `get_certificates_as_lazy`) except for the extra `_as_lazy`, so I suspect the body of this method is meant to be something else entirely and the generator tripped over its shoelaces. I'm not familiar enough with this codebase to propose a fix, but this looks like a relatively straightforward pattern that's gone awry.

**Comment by kairu-ms** [comment]
Thanks for your feedback. We will fix it soon

**Comment by kairu-ms** [comment]
Fixed in version [0.18.1](https://rubygems.org/gems/azure_key_vault/versions/0.18.1)

**Comment by jstange** [comment]
Swapping in the `_async` was one of the things I tried in my blind fumbling, but that returns a `Promise`, which doesn't know what `get_all_items` is:

```
NoMethodError: undefined method `get_all_items' for #<Concurrent::Promise:0x000000000a329a88>
["/var/rubies/ruby-2.7.2/lib/ruby/gems/2.7.0/gems/azure_key_vault-0.18.1/lib/7.1/generated/azure_key_vault/key_vault_client.rb:11125:in `get_certificates_as_lazy'",
 "/var/rubies/ruby-2.7.2/lib/ruby/gems/2.7.0/gems/azure_key_vault-0.18.1/lib/7.1/generated/azure_key_vault/key_vault_client.rb:131:in `get_certificates'"
 ```
 
I haven't used Concurrent before, but trawling around the codebase for other uses of it I think this wants to be something like:
 
 ```
 def get_certificates_as_lazy(vault_base_url, maxresults:nil, include_pending:nil, custom_headers:nil)
       first_page = get_certificates_as_lazy_async(vault_base_url, maxresults:maxresults, include_pending:include_pending, custom_headers:custom_headers)
      first_page.value!
      first_page.get_all_items
end
```

...if I do that, I get... well, a completely new, distinct SDK bug for which I'll open a separate issue in a moment. If I fix that new bug manually, my above modifications yield sensible behavior from `get_certificates_as_lazy` (I get a mundane-looking permission error from the REST API that's almost certainly my fault).
